File: src/commonMain/kotlin/VideoReader.kt
import kotlin.time.Duration
import kotlin.time.ExperimentalTime

interface VideoReader {
    suspend fun majorRevision(): Int

    suspend fun  videoHeader(): VideoHeader

    @ExperimentalTime
    suspend fun videoDuration(): Duration

    @ExperimentalTime
    suspend fun getMediaPagesInRange(start: Duration, length: Duration): Array<ReadableMediaPage>

    suspend fun getVideoResource(resourceId: String): ReadableVideoResource
}

File: lib/src/video_reader.dart
import 'package:npxl_video/generated/npxl_video.pb.dart';
import 'package:npxl_video/src/readable_media_page.dart';
import 'package:npxl_video/src/readable_video_resource.dart';

abstract class VideoReader {
  Future<int> majorRevisionCode();
  Future<VideoHeader> videoHeader();
  Future<Duration> videoDuration();
  Future<Iterable<ReadableMediaPage>> getMediaPagesInRange(
      Duration start, Duration length);
  Future<ReadableVideoResource> getVideoResource(String resourceId);
}
